crates/network-primitives/src/traits.rs

```diff
@@ -107,6 +124,25 @@ impl<T: TransactionResponse> TransactionResponse for WithOtherFields<T> {
     fn input(&self) -> &Bytes {
         self.inner.input()
     }
+
+    fn gas_price(tx: &impl alloy_consensus::Transaction, base_fee: Option<u64>) -> u128 {
+        T::gas_price(tx, base_fee)
+    }
+
+    fn max_fee_per_gas(tx: impl alloy_consensus::Transaction) -> Option<u128> {
+        T::max_fee_per_gas(tx)
+    }
+
+    fn fill(
+        signed_tx: Signed<impl alloy_consensus::Transaction>,
+        signer: Address,
+        block_hash: Option<B256>,
+        block_number: Option<u64>,
+        base_fee: Option<u64>,
+        transaction_index: Option<usize>,
+    ) -> Self {
```

Comment:
this function is problematic because this now effectively freezes what additional fields can be included in a tx response.

Reply:
using `TransactionInfo` isn't possible due to circular dep, should I move `TransactionInfo` into this crate?

Reply:
I think this would be reasonable actually